code/lib/cli/src/generators/baseGenerator.ts

```diff
@@ -260,20 +263,60 @@ export async function baseGenerator(
 
   await configurePreview({ frameworkPreviewParts, storybookConfigFolder, language, rendererId });
 
-  const babelDependencies =
-    addBabel && builder !== CoreBuilder.Vite
-      ? await getBabelDependencies(packageManager, packageJson)
-      : [];
-  const isNewFolder = !files.some(
-    (fname) => fname.startsWith('.babel') || fname.startsWith('babel') || fname === 'package.json'
-  );
-  if (isNewFolder) {
-    await generateStorybookBabelConfigInCWD();
+  const depsToInstall = [...versionedPackages];
+
+  // Add basic babel config for a select few frameworks that need it, if they do not have a babel config file already
+  if (builder !== CoreBuilder.Vite && !skipBabel) {
+    const frameworksThatNeedBabelConfig = [
+      '@storybook/react-webpack5',
+      '@storybook/vue-webpack5',
+      '@storybook/vue3-webpack5',
+      '@storybook/html-webpack5',
+      '@storybook/web-components-webpack5',
+    ];
+    const needsBabelConfig = frameworkPackages.find((pkg) =>
+      frameworksThatNeedBabelConfig.includes(pkg)
+    );
+    const hasNoBabelFile = !files.some(
+      (fname) => fname.startsWith('.babel') || fname.startsWith('babel')
+    );
+
+    if (hasNoBabelFile && needsBabelConfig) {
+      const isTypescript = language !== SupportedLanguage.JAVASCRIPT;
+      const isReact = rendererId === 'react';
+      depsToInstall.push(
+        ...getBabelPresets({
+          typescript: isTypescript,
+          jsx: isReact,
+        })
+      );
+      await writeBabelConfigFile({
+        typescript: isTypescript,
+        jsx: isReact,
+      });
+    }
   }
 
-  const depsToInstall = [...versionedPackages, ...babelDependencies];
+  try {
+    if (process.env.CI !== 'true') {
```

Comment:
In all other places in the code we just check for `if(process.env.CI)`

Reply:
If we do that, the if condition will be matched even if users use `CI=false`, as the value of CI will become `"false"`, and therefore will be truthy 🤔